src/key_info_managers/mod.rs

```diff
@@ -76,6 +123,108 @@ impl KeyTriple {
     }
 }
 
+impl fmt::Display for KeyTriple {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "KeyTriple: app_name=\"{}\" provider_id={} key_name=\"{}\"",
+            self.app_name, self.provider_id, self.key_name
+        )
+    }
+}
+
+impl TryFrom<KeyIdentity> for KeyTriple {
+    type Error = &'static str;
+
+    fn try_from(key_identity: KeyIdentity) -> ::std::result::Result<Self, Self::Error> {
+        let provider_id = match key_identity.provider.uuid.as_str() {
+            crate::providers::core::Provider::PROVIDER_UUID => {
+                Ok::<ProviderId, Self::Error>(ProviderId::Core)
+            }
+            #[cfg(feature = "cryptoauthlib-provider")]
+            crate::providers::cryptoauthlib::Provider::PROVIDER_UUID => {
+                Ok::<ProviderId, Self::Error>(ProviderId::CryptoAuthLib)
+            }
+            #[cfg(feature = "mbed-crypto-provider")]
+            crate::providers::mbed_crypto::Provider::PROVIDER_UUID => {
+                Ok::<ProviderId, Self::Error>(ProviderId::MbedCrypto)
+            }
+            #[cfg(feature = "pkcs11-provider")]
+            crate::providers::pkcs11::Provider::PROVIDER_UUID => {
+                Ok::<ProviderId, Self::Error>(ProviderId::Pkcs11)
+            }
+            #[cfg(feature = "tpm-provider")]
+            crate::providers::tpm::Provider::PROVIDER_UUID => {
+                Ok::<ProviderId, Self::Error>(ProviderId::Tpm)
+            }
+            #[cfg(feature = "trusted-service-provider")]
+            crate::providers::trusted_service::Provider::PROVIDER_UUID => {
+                Ok::<ProviderId, Self::Error>(ProviderId::TrustedService)
+            }
+            _ => Err("Provider does not exist"),
```

Comment:
Maybe something like "Provider not recognized", as it could be that the provider "exists", it just wasn't compiled in. Also, it's worth adding the UUID in there, just to make it easier to debug.